packages/block-editor/src/components/inserter/block-patterns-explorer/explorer.js

```diff
@@ -2,33 +2,69 @@
  * WordPress dependencies
  */
 import { Modal } from '@wordpress/components';
-import { useState } from '@wordpress/element';
+import { useState, useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { usePrevious } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import PatternExplorerSidebar from './sidebar';
 import PatternList from './patterns-list';
+import { usePatternsCategories } from '../block-patterns-tab';
+import { store as blockEditorStore } from '../../../store';
 
-function PatternsExplorer( { initialCategory, patternCategories } ) {
-	const [ filterValue, setFilterValue ] = useState( '' );
+function PatternsExplorer( { initialCategory, rootClientId } ) {
+	const [ searchValue, setSearchValue ] = useState( '' );
+	const [ patternSourceFilter, setPatternSourceFilter ] = useState( 'all' );
+	const patternSyncFilter = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		const settings = getSettings();
+		return settings.patternsSyncFilter || 'all';
+	}, [] );
 	const [ selectedCategory, setSelectedCategory ] = useState(
 		initialCategory?.name
 	);
+
+	const previousSyncFilter = usePrevious( patternSyncFilter );
+
+	// If the sync filter changes, we need to select the "All" category to avoid
+	// showing a confusing no results screen.
+	useEffect( () => {
+		if ( patternSyncFilter && patternSyncFilter !== previousSyncFilter ) {
+			setSelectedCategory( initialCategory?.name );
+		}
+	}, [
+		patternSyncFilter,
+		previousSyncFilter,
+		patternSourceFilter,
+		initialCategory?.name,
+	] );
```

Comment:
Nit: Could we rewrite this without the `useEffect`? Or even better if we could pass `setSelectedCategory` down and call it when the sync filter changes.

Reply:
Changed.

Reply:
yeh it would be nice to do that, but had a look and plumbing through the onCategorySelect  method from modal.js file and using that causes the whole component tree to rerender and the modal gets closed. But would be great to refactor this component tree at some point to allow away for these changes to be handled in the relevant event handlers